integration/aggregate_documents_compat_test.go

```diff
@@ -92,6 +95,12 @@ func testAggregateStagesCompatWithProviders(t *testing.T, providers shareddata.P
 				pipeline = append(pipeline, bson.D{{"$sort", bson.D{{"_id", 1}}}})
 			}
 
+			opts := new(options.AggregateOptions)
+
+			if tc.maxTime != nil {
+				opts.SetMaxTime(pointer.GetDuration(tc.maxTime))
```

Comment:
```suggestion
				opts.SetMaxTime(*tc.maxTime)
```

We already know it is not nil